lib/aws/client.ex

```diff
@@ -194,12 +194,97 @@ defmodule AWS.Client do
     %{client | http_client: http_client}
   end
 
+  @doc """
+  Makes a HTTP request using the specified client.
+
+  `opts` may contain the keyword `:enable_retries?` that enables request retries on known
+  errors such as 500s.
+  The keyword `:retry_opts` can further supply the arguments: `:max_retries`, `:base_sleep_time` &
+  `:cap_sleep_time` that control the retry mechanism. This uses exponential backoff with jitter to
+  sleep in between retry attempts. Default values are:
+  `retry_opts: [max_retries: 10, base_sleep_time: 5, cap_sleep_time: 5_000]`
```

Comment:
We usually have a `Options` section to describe each option

```suggestion
  ## Options
  
    * `:retry_opts` - the options to configure retries in case of errors. This uses exponential backoff with jitter
      to sleep in between retry attempts. They are the following:
      * `:max_retries` - the maximum number of retries. Use `0` to disable retries. Default is `10`.
      * `:base_sleep_time` - the base sleep time in milliseconds. Defaults to `5`.
      * `:cap_sleep_time` - the minimum sleep time in milliseconds for each retry. Defaults to `5_000`.
```

I'm not sure if the indentation is correct.

Reply:
I rewrote it based on your suggestion, and the moduledoc. So hopefully it aligns better now!